huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_sql_control_rule.go

```diff
@@ -0,0 +1,484 @@
+	}
+
+	parts := strings.SplitN(d.Id(), "/", 4)
+	if len(parts) != 4 {
+		return diag.Errorf("invalid id format, must be <instance_id>/<node_id>/<sql_type>/<pattern>")
+	}
+	instanceID := parts[0]
+	nodeId := parts[1]
+	sqlType := parts[2]
+	pattern := parts[3]
+
+	getGaussDBSqlControlRulePath := getGaussDBSqlControlRuleClient.Endpoint + getGaussDBSqlControlRuleHttpUrl
+	getGaussDBSqlControlRulePath = strings.ReplaceAll(getGaussDBSqlControlRulePath, "{project_id}",
+		getGaussDBSqlControlRuleClient.ProjectID)
+	getGaussDBSqlControlRulePath = strings.ReplaceAll(getGaussDBSqlControlRulePath, "{instance_id}", instanceID)
+
+	getGaussDBSqlControlRulePath += buildGetGaussDBSqlControlRuleQueryParams(nodeId)
+
+	getGaussDBSqlControlRuleOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+	getGaussDBSqlControlRuleResp, err := getGaussDBSqlControlRuleClient.Request("GET",
+		getGaussDBSqlControlRulePath, &getGaussDBSqlControlRuleOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+			"error retrieving GaussDB MySQL SQL control rule")
+	}
+
+	getGaussDBSqlControlRuleRespBody, err := utils.FlattenResponse(getGaussDBSqlControlRuleResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	maxConcurrency, err := flattenGetGaussDBSqlControlRuleResponseBody(getGaussDBSqlControlRuleRespBody, sqlType, pattern)
+	if err != nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", instanceID),
+		d.Set("node_id", nodeId),
+		d.Set("sql_type", sqlType),
+		d.Set("pattern", pattern),
+		d.Set("max_concurrency", maxConcurrency),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetGaussDBSqlControlRuleQueryParams(nodeId string) string {
+	return fmt.Sprintf("?node_id=%v", nodeId)
+}
+
+func flattenGetGaussDBSqlControlRuleResponseBody(resp interface{}, sqlType, pattern string) (int, error) {
+	if resp == nil {
+		return 0, fmt.Errorf("error get GaussDB MySQL SQL control rule")
+	}
+	filterRules := utils.PathSearch("sql_filter_rules", resp, make([]interface{}, 0))
+	for _, filterRule := range filterRules.([]interface{}) {
+		resSQLType := utils.PathSearch("sql_type", filterRule, "").(string)
+		if sqlType == resSQLType {
+			resPatterns := utils.PathSearch("patterns", filterRule, make([]interface{}, 0))
+			for _, v := range resPatterns.([]interface{}) {
+				resPattern := utils.PathSearch("pattern", v, "").(string)
+				if resPattern == pattern {
+					maxConcurrency := utils.PathSearch("max_concurrency", v, 0).(float64)
+					return int(maxConcurrency), nil
+				}
+			}
+		}
+	}
+	return 0, fmt.Errorf("error get GaussDB MySQL SQL control rule")
+}
+
+func resourceGaussDBSqlControlRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Seems like update function is same to create function. How about just keep one function for both create and update function

Reply:
done